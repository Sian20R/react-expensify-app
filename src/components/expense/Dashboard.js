import React from 'react';
import ExpenseList from './../expense/List';
import ExpenseListFilters from './../expense/ListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);
export default ExpenseDashboardPage;