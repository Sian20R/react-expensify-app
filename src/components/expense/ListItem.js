import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const ExpenseItem = ({id, description, amount, createdAt}) => (
    <div>
      <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
      <p>{amount} - {createdAt}</p>    
     
    </div>
);

const mapStateToProps = (state) => {
  return {
      expenses: state.expenses
  };
};

export default connect(mapStateToProps)(ExpenseItem);