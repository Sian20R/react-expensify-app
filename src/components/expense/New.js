import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../../actions/expenses';
import ExpenseForm from '../expense/Form';

export class NewExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.newExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <h1>New Expense</h1>
                <ExpenseForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    } 
};

const mapDispatchToProps = (dispatch) => ({
    newExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(NewExpensePage);