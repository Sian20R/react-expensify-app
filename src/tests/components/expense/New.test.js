import React from 'react';
import { shallow } from 'enzyme';
import { NewExpensePage } from '../../../components/expense/New';
import expenses from '../../fixtures/expenses';

let newExpense, history, wrapper;

beforeEach(() => {
    newExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<NewExpensePage newExpense={newExpense} history={history} />)
});

test('should render NewExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(newExpense).toHaveBeenLastCalledWith(expenses[1]);
});