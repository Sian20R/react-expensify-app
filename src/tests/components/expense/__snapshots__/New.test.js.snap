// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NewExpensePage correctly 1`] = `
<div>
  <h1>
    New Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
